### PR TITLE
bpo-23878: Remove the key parameter of _Py_FindEnvConfigValue()

### PR DESCRIPTION
**Pull request by ZackerySpytz**
Suggested by @vstinner in GH-15424.

https://bugs.python.org/issue23878

### CODE REVIEW

File: Python/pathconfig.c [hunk 1]
```diff
@@ -734,8 +734,7 @@ _PyPathConfig_ComputeSysPath0(const PyWideStringList *argv, PyObject **path0_p)
 /* Search for a prefix value in an environment file (pyvenv.cfg).
```
Comment (vstinner) [inline Python/pathconfig.c:734]: It doesn't search "for a prefix". It searchs for the the line "home=<path>" and return `<path>`. It tolerates whitespaces around "=".

**Comment by vstinner** [comment]
IMO it's fine to reuse https://bugs.python.org/issue23878 for this cleanup change. It gives more context about this change, than no bpo number.

**Review by zooba — CHANGES_REQUESTED** [review]
If we're going to remove the key argument, we should also rename the function to reflect that it only finds `home` somehow.

Personally, I see no benefit here, and we may as well leave it alone. But if others think it's cleaner to have `_Py_FindEnvConfigHomeValue` instead of specifying `L"home"` in the call, I'm not going to fight it.

**Comment by csabella** [comment]
@ZackerySpytz, please resolve the merge conflicts and address the code review.  If this change is no longer relevant, please close the PR.  Thanks!

**Comment by csabella** [comment]
@ZackerySpytz ping

**Comment by iritkatriel** [comment]
https://bugs.python.org/issue23878 is closed. What is that status of this PR?

**Comment by vstinner** [comment]
This PR is still relevant, but it's outdated. @ZackerySpytz: can you please rebase it on the main branch? Or create a new PR.

**Comment by kumaraditya303** [comment]
Closing as it is outdated and bpo is closed, if required a new PR can be opened or this can reopened.

**Pull request closed without merging**